datastore/modelassertmanager.go

```diff
@@ -36,40 +36,42 @@ const createModelAssertTableSQL = `
 		gadget           varchar(60) not null,
 		kernel           varchar(60) not null,
 		store            varchar(60),
+		required_snaps   text default '',
 		created          timestamp default current_timestamp,
-		modified         timestamp default current_timestamp
+		modified         timestamp default current_timestamp,
 	)
 `
 const createModelAssertSQL = `
 INSERT INTO modelassertion 
-(model_id,keypair_id,series,architecture,revision,gadget,kernel,store) 
-VALUES ($1,$2,$3,$4,$5,$6,$7,$8) 
+(model_id,keypair_id,series,architecture,revision,gadget,kernel,store,required_snaps) 
+VALUES ($1,$2,$3,$4,$5,$6,$7,$8,$9) 
 RETURNING id`
 
 const updateModelAssertSQL = `
 UPDATE modelassertion
-SET model_id=$2, keypair_id=$3, series=$4, architecture=$5, revision=$6, gadget=$7, kernel=$8, store=$9, modified=$10
+SET model_id=$2, keypair_id=$3, series=$4, architecture=$5, revision=$6, gadget=$7, kernel=$8, store=$9, modified=$10, required_snaps=$11 
 WHERE id=$1`
 
 const getModelAssertSQL = `
-SELECT id,model_id,keypair_id,series,architecture,revision,gadget,kernel,store,created,modified
+SELECT id,model_id,keypair_id,series,architecture,revision,gadget,kernel,store,required_snaps,created,modified
 FROM modelassertion
 WHERE model_id=$1
 `
 
 // ModelAssertion holds the model assertion details in the local database
 type ModelAssertion struct {
-	ID           int       `json:"id"`
-	ModelID      int       `json:"model_id"`
-	KeypairID    int       `json:"keypair_id"`
-	Series       int       `json:"series"`
-	Architecture string    `json:"architecture"`
-	Revision     int       `json:"revision"`
-	Gadget       string    `json:"gadget"`
-	Kernel       string    `json:"kernel"`
-	Store        string    `json:"store"`
-	Created      time.Time `json:"created"`
-	Modified     time.Time `json:"modified"`
+	ID            int       `json:"id"`
+	ModelID       int       `json:"model_id"`
+	KeypairID     int       `json:"keypair_id"`
+	Series        int       `json:"series"`
+	Architecture  string    `json:"architecture"`
+	Revision      int       `json:"revision"`
+	Gadget        string    `json:"gadget"`
+	Kernel        string    `json:"kernel"`
+	Store         string    `json:"store"`
+	RequiredSnaps string    `json:"required-snaps"`
```

Comment:
won't that dash cause problems in json?. You'd better use 'required_snaps' i think

Reply:
The js code actually handles that in this case. I've wanted to switch to underscores over time, but this change was needed quickly.